spec/controllers/menus_controller_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+RSpec.describe MenusController, type: :controller do
+  render_views
+
+  let!(:user) { create(:user) }
+  let(:product_1) { create(:product, name: 'Product 1') }
+  let(:product_2) { create(:product, name: 'Product 2') }
+  let(:product_3) { create(:product, name: 'Product 3') }
+
+  let!(:menu_1) { create(:menu, product_ids: [product_1.id, product_2.id]) }
+  let!(:menu_2) { create(:menu, product_ids: [product_3.id]) }
+
+  before { sign_in user }
+    
+  describe 'GET index' do
+    before { get :index }
+
+    specify do
+      expect(response).to have_http_status(200) 
+      expect(response).to render_template(:index) 
+      expect(assigns(:menus)).to eq [menu_1, menu_2]
+    end
+  end
+
+  describe 'GET new' do
+    before { get :new }
+
+    specify do
+      expect(response).to have_http_status(200) 
+      expect(response).to render_template(:new)
+    end
+  end 
+
+  describe 'POST create' do
+
+    let(:params) do
+    {
+      menu_1: {
+        menu_date: '15/05/2019',
+        product_ids: [2,3],
+        created_by: 1
+      }
+    }
+    end
+
+    context 'success' do
+      specify do
+        expect do
+          post :create, params: params
+        end.to change(Menu, :count).by(1)
+        product = Menu.first
+        expect(menu_1.menu_date).to eq '15/05/2019'
+        expect(menu_1.products.first).to eq [2]
+        expect(menu_1.created_by).to eq 1
```

Comment:
`        expect(menu.user.email).to eq 'test@test.com'
`

Reply:
create_by cần xác định chính xác user update, ở đây ta có thể dùng user email, vì vậy user đang login là user có email `test@test.com`, nên chỉ cần kiểm tra email là được hoặc cũng có thể kiểm tra bất kỳ field nào của user.